client/src/app/product/[productid]/page.tsx

```
@@ -10,28 +10,34 @@
-  fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/api/category/${category}`)
-    .then(response => response.json())
-    .then(data => setCategoryName(data.data.name))
-    .catch(error => console.error('Error:', error));
 
-  const { productid } = use(params);
+  useEffect(() => {
+    if (category) {
+      fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/api/category/${category}`)
+        .then(response => response.json())
+        .then(data => {
+          if (data && data.data) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)